ctwdtw/GitHubAPIPractice
# merge dev to main

- duplicate UserProfileUIComposer as UserDetailUIComposer
- rename `UserDetailProfile` to `UserDetail`
- rename `RemoteUserDetailProfileLoader` to `RemoteUserDetailLoader`
- delete unnecessary userProfile wording in `UserDetailUIComposer`
- duplicate `UserProfileCell` as `UserDetailCell`
- `RemoteUserDetailLoader` delivers one instance of `UserDetail`
- `UserDetailLoader` returns one `UserDetail` instance
- fix snapshot build error
- add `UserDetailSnapshotTests`
- extract `SiteAdminView` from `SiteAdminCell`
- prepare refactoring
- extract layout code for `SiteAdminCell`
- extract layout for `DetailFieldCell`
- move `SiteAdminCell`, `DetailFieldCell` to prod. code
- update snapshot for `SiteAdminCell`
- add snapshot for `DetailFieldCell` and user detail list
- adjust folder structure for snapshot tests
- render `UserDetail` fields on `ListViewController`
- request image when avatarView is visible
- display image loading indicator while loading image
- render loaded image on image data loading complete
- rename variable
- show retry action view on image data loading complete with error
- show retry action view when loaded invalid image data
- retry image loading action on tap retry action view
- add snapshot for `AvatarCell` when should retry image
- cancel avatar image loading when `ListViewController` de-int or when avatar view is not visible anymore
- notify selection handler on profile selected
- embed `ListViewController` in `UINavigation` controller
- move `navigationController` out as instance variable of `SceneDelegate` because we want o use it to push other view controller
- `showDetail` push user detail list
- make test case more readable
- fix missing icon for `SiteAdminCell`
- remove default label text color of `DetailFieldCell`
- fix text color for `DetailFieldCell` inside `BlogCellController`
- lower priority of `AvatarCell.stackView` to resolve constraint conflict
- add simple ui test to verify navigation and pagination
- add page object model and helpers to enhance test case readability
- move ui test to another scheme
- fix a layout bug due to reuse of table view cell
